src/main/java/hudson/remoting/Channel.java

```diff
@@ -327,6 +327,12 @@
     /*package*/ @Nonnull final ClassFilter classFilter;
 
     /**
+     * Indicates that close of the channel has been requested.
+     * After it it does not make sense to execute new user-space commands.
+     */
+    private boolean closeRequested = false;
```

Comment:
Yes, race condition. `outClosed` is set within the `synchronize(this)` block. If there is a deadlock or delay, the commands will be accepted even if the `close()` command has been called somewhere